### PR TITLE
Blockstore retrieval integration tests

### PR DESCRIPTION
**Pull request by monsieurleberre**
cf testcases 864 and 865

### CODE REVIEW

File: src/Stratis.Bitcoin.IntegrationTests/BlockStore/RetrieveFromBlockStoreSpecification.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            And(the_node_is_synced);
+
+            When(trying_to_retrieve_the_blocks_from_the_blockstore);
+
+            Then(real_blocks_should_be_retrieved);
+            Then(made_up_blocks_should_not_be_retrieved);
+        }
+
+        [Fact]
+        public void A_transaction_can_be_retrieved_by_its_uint256_identifier()
```
Comment (ferdeen) [inline src/Stratis.Bitcoin.IntegrationTests/BlockStore/RetrieveFromBlockStoreSpecification.cs]: Hi, could the real and made up blocks be separate tests ? The unit test name `A_block_can_be_retrieved_by_its_uint256_identifier()` suggests you are testing for a good block only.  

Or change the name to `A_block_can_be_retrieved_by_its_uint256_identifier_and_a_made_up_block_rejected()` ?

File: src/Stratis.Bitcoin.IntegrationTests/BlockStore/RetrieveFromBlockStoreSteps.cs [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        private IEnumerable<uint256> retrievedBlockHashes;
+        private CoreNode transactionNode;
+        private readonly Money transferAmount = Money.COIN * 2;
+        private Transaction transaction;
+        private HdAddress receiverAddress;
+        private uint256 blockWithTransactionId;
+        private Transaction retrievedTransaction;
+        private uint256 madeUpTransactionId;
+        private Transaction wontRetrieveTransaction
+            ;
```
Comment (ferdeen) [inline src/Stratis.Bitcoin.IntegrationTests/BlockStore/RetrieveFromBlockStoreSteps.cs]: A lost semi-colon :)

File: src/Stratis.Bitcoin.IntegrationTests/BddTemplate.DotSettings [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+		public void $a_result_should_be_observed$()
+		{
+			
+		}
+
+		public void $it_should_satisfy_our_specification$()
+		{
+			"".Should().NotBeNullOrWhiteSpace();
+		}
+    }
```
Comment (ferdeen) [inline src/Stratis.Bitcoin.IntegrationTests/BddTemplate.DotSettings]: Nice one!

File: src/Stratis.Bitcoin.IntegrationTests/BlockStore/RetrieveFromBlockStoreSpecification.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+// Disable warnings about "this" qualifier to make the Specification more readable
+// ReSharper disable ArrangeThisQualifier
+
+namespace Stratis.Bitcoin.IntegrationTests.BlockStore
+{
+    public partial class RetrieveFromBlockStoreSpecification : BddSpecification
+    {
+        [Fact]
+        public void A_block_can_be_retrieved_by_its_uint256_identifier()
```
Comment (ferdeen) [inline src/Stratis.Bitcoin.IntegrationTests/BlockStore/RetrieveFromBlockStoreSpecification.cs]: Does the type actually matter in the test name (unit256) ? 

And split the unit test or rename - see my next comment below.

File: src/Stratis.Bitcoin.IntegrationTests/BddTemplate.DotSettings (hunk 3 above)
Comment (monsieurleberre) [reply src/Stratis.Bitcoin.IntegrationTests/BddTemplate.DotSettings]: So you start with a failing test :)

File: src/Stratis.Bitcoin.IntegrationTests/BlockStore/RetrieveFromBlockStoreSpecification.cs (hunk 1 above)
Comment (monsieurleberre) [reply src/Stratis.Bitcoin.IntegrationTests/BlockStore/RetrieveFromBlockStoreSpecification.cs]: OK sorry, I think I will rename unless someone insists on separating the tests, I think the set up of these tests is so long that in that case grouping a trivial test with a normal one is valid... Thanks!

**Comment by bokobza** [comment]
Looks like this PR includes changes from another PR.

File: src/Stratis.Bitcoin.IntegrationTests/BlockStore/RetrieveFromBlockStoreSpecification.cs (hunk 1 above)
Comment (ferdeen) [reply src/Stratis.Bitcoin.IntegrationTests/BlockStore/RetrieveFromBlockStoreSpecification.cs]: Yes on thought so to.

**Comment by monsieurleberre** [comment]
@bokobza yes it includes the changes I made with OP_RETURN, but the only common thing is the support for async BDD testing. If you prefer I will isolate the 2 commits.

**Comment by bokobza** [comment]
Thanks mate!

**Pull request merged**